priam/src/main/java/com/netflix/priam/resources/CassandraAdmin.java

```diff
@@ -138,75 +141,45 @@ public Response cassRing(@PathParam("id") String keyspace) throws IOException, I
     @GET
     @Path("/flush")
     public Response cassFlush() throws IOException, InterruptedException, ExecutionException {
-        JMXConnectorMgr connMgr = null;
+        JSONObject rootObj = new JSONObject();
+
         try {
-            connMgr = new JMXConnectorMgr(config);
-        } catch (IOException | InterruptedException e) {
-            JSONObject rootObj = new JSONObject();
+            flush.execute();
+            rootObj.put("Flushed", true);
+            return Response.ok().entity(rootObj).build();
+        } catch (Exception e) {
             try {
                 rootObj.put("status", "ERRROR");
-                rootObj.put("component", "jmxconnector");
                 rootObj.put("desc", e.getLocalizedMessage());
             } catch (Exception e1) {
-                return Response.status(503).entity("JMXConnector error")
+                return Response.status(503).entity("FlushError")
                         .build();
             }
             return Response.status(503).entity(rootObj)
                     .build();
         }
+    }
+
+    @GET
+    @Path("/compact")
+    public Response cassCompact() throws IOException, ExecutionException, InterruptedException {
+        JSONObject rootObj = new JSONObject();
 
-        List<String> flushed = null;
         try {
-            IClusterManagement task = new Flush(this.config, connMgr);
-            flushed = task.execute();
+            compaction.execute();
```

Comment:
Do you care or take care of scheduled compaction is running and calling this REST endpoint causes to start compaction while there is one already running.

Reply:
This is a REST endpoint where it is doing something requested by "admin". I don't think we should check (or change existing behavior) for any existing running compaction.

Reply:
How would you avoid admin issuing REST command while there is already a scheduled compaction running? One suggestion is to either queue the task or reject the REST endpoint request stating that there is already a running compaction and admin can retry at a later point in time.

Reply:
Resolved in IClusterManagement.